File: src/main/java/ie/hunt/sb/hello/HelloController.java
package ie.hunt.sb.hello;


import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class HelloController {

    @GetMapping("/")
    public String index() {
        return "Greetings from Spring Boot!";
    }


    @GetMapping("/{{name}}")
    public String greet(String name) {
        //TODO: broken .. how do we map path params
        //TODO: broken .. no test cases .. how do we write a test-case??
        return "Greetings from Spring Boot:" + name;
    }

}

File: src/main/kotlin/ie/hunt/sb/hello/HelloController.kt
package ie.hunt.sb.hello

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.web.bind.annotation.RestController

@RestController
class HelloController(var config: IConfig) {

    @GetMapping("/", produces = ["text/plain"])
    @ResponseBody
    fun index(): String {
        return "Hello World" + "\nSupported endpoints: " + "\n/hello/{name}"
    }

    /**
     * Map the path variable {{name}} and use it in the response
     * https://www.baeldung.com/spring-pathvariable#specifying-the-request-parameter-name
     */
    @GetMapping("/hello/{name}", produces = ["text/plain"])
    @ResponseBody
    fun greet(@PathVariable("name") name: String): String {
        return "Hello $name"
    }


    @GetMapping("/git_sha", produces = ["text/plain"])
    @ResponseBody
    fun version () : String {
        return GVersion.GIT_SHA

    }
}
